packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -105,38 +104,27 @@ const sortNodes = (a, b) => {
 const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selectionMode, children }) => {
   useLifecycleLogging('SceneHierarchyDataProvider');
   const sceneComposerId = useSceneComposerId();
-  const { updateSceneNodeInternal, isEditing } = useStore(sceneComposerId)((state) => state);
-  const selectedSceneNodeRef = useStore(sceneComposerId)((state) => state.selectedSceneNodeRef);
-  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
-  const getObject3DBySceneNodeRef = useStore(sceneComposerId)((state) => state.getObject3DBySceneNodeRef);
+  const {
+    document,
+    selectedSceneNodeRef,
+    getSceneNodeByRef,
+    setSelectedSceneNodeRef,
+    updateSceneNodeInternal,
+    getObject3DBySceneNodeRef,
+    setCameraTarget,
+    removeSceneNode,
+    isEditing,
+  } = useStore(sceneComposerId)((state) => state);
```

Comment:
using individual hooks is a performance improvement from https://github.com/awslabs/iot-app-kit/pull/283/files#diff-9eae39af074be35fac13ff65406838d84bce07db7e48db1e4654f15a227e23d9